languages/csharp/exercises/concept/inheritance/.meta/Example.cs

```diff
@@ -0,0 +1,81 @@
+abstract class Character
+{
+    private int _hitPoints;
+
+    protected Character(int hitPoints)
+    {
+        _hitPoints = hitPoints;
+    }
+
+    protected abstract int Damage();
+
+    public void Attack(Character target)
+    {
+        if (Stunned())
+        {
+            return;
+        }
+
+        target._hitPoints -= Damage();
+    }
+
+    public bool Stunned()
+    {
+        return _hitPoints <= 0;
+    }
+
+    public override string ToString()
+    {
+        return $"HP: {_hitPoints}";
+    }
+}
+
+class Wizard : Character
+{
+    private bool spellPrepared;
+
+    public Wizard() : base(20)
+    {
+    }
+
+    public void PrepareSpell()
+    {
+        spellPrepared = true;
+    }
+
+    protected override int Damage()
```

Comment:
What do you think about renaming this to `AttackDamage`? I'm just thinking a Wizard doesn't have a Damage as much as they cause damage when they attack. It's a really minor point though.

Reply:
Well, I think in most games it would still be damage, regardless of the fact if any direct damage is caused. I can see your point though. One benefit of `AttackDamage` is that it ties it to the `Attack` method. That is also a disadvantage, as there can be confusion between which of the two methods to work on by the student due to very similar names.

I'm leaning towards _not_ changing it, but lets ask a third opinion :) @SleeplessByte what do you think of the above suggestion?

Reply:
The domain would actually prefer Attack or Physical, and Magic, or Strength and Intelligence, or a different permutation of those, over damage, as they are short for stat that is the main component in determining damage when attacking.

In other words, I think damage should realistically never be a property of any actor such as a wizard or monster; but rather a standalone function that also takes into account the defense and evasion of the recipient of the damage.

The attack action should probably live on the attacker. But it alone can not determine the damage; it needs the thing it attacks.

If you model it like that, the need for this method goes away.